deepchem/models/torch_models/prot_bert.py

```diff
@@ -0,0 +1,107 @@
+
+    ProtBERT model is based on BERT architecture and the current implementation
+    supports only MLM pretraining and classification mode, as described by the
+    authors in HuggingFace[2]. For classfication we currently only support
+    Logistic regression and a simple Feed forward neural network.
+
+
+    Parameters
+    ----------
+    task: str
+        The task defines the type of learning task in the model. The supported tasks are
+         - `mlm` - masked language modeling commonly used in pretraining
+         - `classification` - use it for classification tasks
+    model_path: str
+        Path to the HuggingFace model
+    n_tasks: int, default 1
+        Number of prediction targets for a multitask learning model
+    cls_name: str
+        The classifier head to use for classification mode.
+
+    Example
+    -------
+    >>> import os
+    >>> import tempfile
+    >>> tempdir = tempfile.mkdtemp()
+
+    >>> # preparing dataset
+    >>> import pandas as pd
+    >>> import deepchem as dc
+    >>> protein = ["MPCTTYLPLLLLLFLLPPPSVQSKV","SSGLFWMELLTQFVLTWPLVVIAFL"]
+    >>> labels = [0,1]
+    >>> df = pd.DataFrame(list(zip(protein, labels)), columns=["protein", "task1"])
+    >>> with dc.utils.UniversalNamedTemporaryFile(mode='w') as tmpfile:
+    ...     df.to_csv(tmpfile.name)
+    ...     loader = dc.data.CSVLoader(["task1"], feature_field="protein", featurizer=dc.feat.DummyFeaturizer())
+    ...     dataset = loader.create_dataset(tmpfile.name)
+
+    >>> # pretraining
+    >>> from deepchem.models.torch_models.prot_bert import ProtBERT
+    >>> pretrain_model_dir = os.path.join(tempdir, 'pretrain-model')
+    >>> model_path = 'Rostlab/prot_bert'
+    >>> pretrain_model = ProtBERT(task='mlm', HG_model_path=model_path, n_tasks=1, model_dir=pretrain_model_dir)  # mlm pretraining
+    >>> pretraining_loss = pretrain_model.fit(dataset, nb_epoch=1)
+    >>> del pretrain_model
+
+    >>> finetune_model_dir = os.path.join(tempdir, 'finetune-model')
+    >>> finetune_model = ProtBERT(task='classification', HG_model_path=model_path, n_tasks=1, model_dir=finetune_model_dir)
+    >>> finetune_model.load_from_pretrained(pretrain_model_dir)
+    >>> finetuning_loss = finetune_model.fit(dataset, nb_epoch=1)
+
+    >>> # prediction and evaluation
+    >>> result = finetune_model.predict(dataset)
+    >>> eval_results = finetune_model.evaluate(dataset, metrics=dc.metrics.Metric(dc.metrics.accuracy_score))
+
+    References
+    ----------
+    .. [1] Elnaggar, Ahmed, et al. "Prottrans: Toward understanding the language of life through self-supervised learning." IEEE transactions on pattern analysis and machine intelligence 44.10 (2021): 7112-7127.
+    .. [2] https://huggingface.co/Rostlab/prot_bert
+
+    """
+
+    def __init__(self,
+                 task: str,
+                 model_path: str = 'Rostlab/prot_bert',
+                 n_tasks: int = 1,
+                 cls_name: str = "LogReg",
+                 **kwargs):
```

Comment:
Add docstrings explaining the parameters

Reply:
Added.